cmake/CPM.cmake

```diff
@@ -42,7 +42,11 @@ if(NOT COMMAND cpm_message)
   endfunction()
 endif()
 
-set(CURRENT_CPM_VERSION 1.0.0-development-version)
+if(DEFINED CPM_VERSION AND CPM_RELEASE)
+  set(CURRENT_CPM_VERSION "${CPM_VERSION}${CPM_DEVELOPMENT}")
+else()
+  set(CURRENT_CPM_VERSION 1.0.0-development-version)
```

Comment:
With the `.git_archival.txt` we have more details on commit version. Should use that information.